Framework/Algorithms/src/CreateWorkspace.cpp

```diff
@@ -144,16 +161,15 @@ void CreateWorkspace::exec() {
   }
 
   // Create the OutputWorkspace
+  auto storageMode = Parallel::fromString(getProperty("ParallelStorageMode"));
+  IndexInfo indexInfo(nSpec, storageMode, communicator());
```

Comment:
I really dislike the `communicator()` argument: This should always be the communicator held by the algorithm, so passing it is just repetitive boild plate, however I could not think of a good way to eliminate this. One option might be to make `Algorithm` a factory for `IndexInfo`, but I think that violates some other design principles.

Ideas or comments?